src/riak_kv_vnode.erl

```diff
@@ -1540,26 +1540,19 @@ prepare_put_existing_object(#state{idx =Idx} = State,
     {ActorId, State2} = maybe_new_key_epoch(Coord, State, OldObj, RObj),
     case put_merge(Coord, LWW, OldObj, RObj, ActorId, StartTime) of
         {oldobj, OldObj} ->
-            {{false, OldObj}, PutArgs, State2};
+            {{false, {OldObj, no_old_object}}, PutArgs, State2};
```

Comment:
Could replace {false, {OldObj, no_old_object}} by {false, OldObj}, changing perform_put to reflect this, as there is never a new object in the false case.